src/model/product.ts

```diff
@@ -7,6 +7,10 @@ export class Product {
   id: number;
   title: string;
   image: string;
-  price: string;
+  price: number;
   description: string;
 }
+
+export class ProductInCart extends Product {
```

Comment:
I think no.  Id is necessary for other feratures too, not only for the cart.

Reply:
yeah but in this way we duplicate the data, this could lead to inconsistencies

Reply:
the product type is wired through the store,routing, etc. They dont need the count at all. It would be nonesense to use null values there,only becasue we want to use there this count,for no reason. It is common to extend types in thypescript, I dont understand why would it be a problem.

Reply:
anyway, it's been fixed with separated type + mapping